Ed/gps path follow documentation

Ezward opened the pull request:
This adds documentation for the path follow template.  I've reorganized the docs a little to separate the deep learning template and the path follow template.  Docs below 'Get Driving' are separated into the deep-learning and path-follow templates.  This structure make sense to me and it will scale when we upgrade the computer-vision template.  Does this make sense to others (that is why I did this as PR, so it can get reviewed and discussed).

- added path follow content to the guide
- moved the deep learning specific guide content into `guide/deep_learning`.
- created a new `guide\get_driving.md` that discusses getting driving using the deep learning or the path follow templates and leads to that specific content.
- added mkdocs-redirect plugin and added redirects so links to the old guide pages `guide/*.md` will be redirected to the new location `guide/deep_learning/*.md`.

DocGarbanzo commented:
@Ezward - did you run the webserver locally? For me a couple of links are not working. You can run it by:
```
conda install mkdocs
mkdocs serve
```
and go to `http://127.0.0.1:8000`. For example from the navigation bar on the lhs I cannot reach: dataset_and_pretrained_models.md, simulator.md, virtual_race_league.md, mobile_app.md.

Ezward commented:
> and go to `http://127.0.0.1:8000`. For example from the navigation bar on the lhs I cannot reach: dataset_and_pretrained_models.md, simulator.md, virtual_race_league.md, mobile_app.md.

Ok, I missed the file that defines the navigation links.  Also, I went back and fixed some broken links that include the .md extension and/or the /docs/ path.  It is very aggravating that links that work in the github repo do _not_ work in the final jekyll html and visa versa.  Oh and PS the file that defines the navigation links does use the .md extension.  WTF.

Ezward commented:
Note I changed the nav just a bit.  I removed the stuff about donkey datasets because it was feeble; where datasets are most useful (the simulator) we don't have any.  I then add Donkey UI to the main nav so it is much more visible.

Ezward commented:
I added the mkdocs-redirects plugin and added redirects for the deep learning content.

DocGarbanzo commented:
Thanks @Ezward - navigation is all working as far as I can see. I've stumbled over the different path behaviours between the native rendering of the md files in my IDE vs what happens under mkdocs as well in the past - that's why I knew what to look out for :-) I bumped super-linter to v4 on master and merged back into your branch, I think its all good.

DocGarbanzo commented on .github/workflows/push-docs.yml:
```diff
@@ -23,5 +23,6 @@ jobs:
           conda info
           conda list
           conda install mkdocs
+          conda install mkdocs-redirects
```
Yep, great, just ran into that problem myself....
